README.md
# VCard PHP class
[![Latest Stable Version](http://img.shields.io/packagist/v/jeroendesloovere/vcard.svg)](https://packagist.org/packages/jeroendesloovere/vcard)
[![License](http://img.shields.io/badge/license-MIT-lightgrey.svg)](https://github.com/jeroendesloovere/vcard/blob/master/LICENSE)
[![Build Status](https://travis-ci.org/jeroendesloovere/vcard.svg?branch=master)](https://travis-ci.org/jeroendesloovere/vcard)
[![Code Coverage](https://scrutinizer-ci.com/g/jeroendesloovere/vcard/badges/coverage.png?b=master)](https://scrutinizer-ci.com/g/jeroendesloovere/vcard/?branch=master)

This VCard PHP class can generate a vCard with some data. When using an iOS device (< iOS 8) it will export as a .ics file because iOS devices don't support the default .vcf files.

## Usage

### Installation

``` json
{
    "require": {
        "jeroendesloovere/vcard": "1.2.*"
    }
}
```

> Add the above in your `composer.json` file when using [Composer](https://getcomposer.org).

### Example

``` php
use JeroenDesloovere\VCard\VCard;

// define vcard
$vcard = new VCard();

// define variables
$lastname = 'Desloovere';
$firstname = 'Jeroen';
$additional = '';
$prefix = '';
$suffix = '';

// add personal data
$vcard->addName($lastname, $firstname, $additional, $prefix, $suffix);

// add work data
$vcard->addCompany('Siesqo');
$vcard->addJobtitle('Web Developer');
$vcard->addEmail('info@jeroendesloovere.be');
$vcard->addPhoneNumber(1234121212, 'PREF;WORK');
$vcard->addPhoneNumber(123456789, 'WORK');
$vcard->addAddress(null, null, 'street', 'worktown', null, 'workpostcode', 'Belgium');
$vcard->addURL('http://www.jeroendesloovere.be');

$vcard->addPhoto(__DIR__ . '/landscape.jpeg');

// return vcard as a string
//return $vcard->getOutput();

// return vcard as a download
return $vcard->download();
```

> [View all examples](/examples/example.php) or check [the VCard class](/src/VCard.php).

**Support for frameworks**

I've created a Symfony Bundle: [VCard Bundle](https://github.com/jeroendesloovere/vcard-bundle)

Usage in for example: Laravel
```php
return Response::make(
    $this->vcard->getOutput(),
    200,
    $this->vcard->getHeaders(true)
);
```

## Documentation

The class is well documented inline. If you use a decent IDE you'll see that each method is documented with PHPDoc.

## Contributing

Contributions are **welcome** and will be fully **credited**.

### Pull Requests

> To add or update code

- **Coding Syntax** - Please keep the code syntax consistent with the rest of the package.
- **Add unit tests!** - Your patch won't be accepted if it doesn't have tests.
- **Document any change in behavior** - Make sure the README and any other relevant documentation are kept up-to-date.
- **Consider our release cycle** - We try to follow [semver](http://semver.org/). Randomly breaking public APIs is not an option.
- **Create topic branches** - Don't ask us to pull from your master branch.
- **One pull request per feature** - If you want to do more than one thing, send multiple pull requests.
- **Send coherent history** - Make sure each individual commit in your pull request is meaningful. If you had to make multiple intermediate commits while developing, please squash them before submitting.

### Issues

> For bug reporting or code discussions.

More info on how to work with GitHub on help.github.com.

## Credits

- [Jeroen Desloovere](https://github.com/jeroendesloovere)
- [All Contributors](https://github.com/jeroendesloovere/vcard/contributors)

## License

The module is licensed under [MIT](./LICENSE.md). In short, this license allows you to do everything as long as the copyright statement stays present.

CHANGELOG.md
1.2.0 (2015-06-01)
--
Improvements:
* You can now add some properties multiple times: email, address, phone number and url

1.1.11 (2015-05-21)
--
Improvements:
* addMedia updated to check if correct file type.

1.1.10 (2015-05-20)
--
Improvements:
* Multiple mailaddresses allowed.
* Chaining integrated to add functions.

1.1.9 (2015-04-21)
--
Improvements:
* getHeaders() is now separate function. So frameworks can use this.
* Fix for iOS 8 to return vcard without calendar wrapper.

1.1.8 (2015-03-09)
--
Bugfixes:
* Fixes $include/$exclude, #27
* Fixes special characters by using external transliterator class.

1.1.7 (2015-03-05)
--
Improvements:
* Images should per default be included in our vcard.

Bugfixes:
* Fix for the ->get() which didn't return anything.

1.1.6 (2015-02-24)
--
Improvements:
* Add line folding, check #16
* Refactored some functions.
* PSR-2-code-styling applied.
* PHPCS applied.

Bugfixes:
* Fix fetching PHOTO elements.

1.1.5 (2015-01-30)
--
Bugfixes:
* Updated the deprecated MIME detection, check #16

1.1.4 (2015-01-22)
--
Improvements:
* PHPUnit Tests added
* Exception is now a separate class.
* Renamed the variables $firstName and $lastName

Bugfixes:
* Filename: Fixed double underscores when no "additional" field was given.

1.1.3 (2015-01-22)
--
Bugfixes:
* Name: Double space when no "additional" field is given. Fixes #8


examples/example.php
<?php

/**
 * VCard generator test - can save to file or output as a download
 *
 * @author Jeroen Desloovere <info@jeroendesloovere.be>
 */

require_once __DIR__ . '/../vendor/autoload.php';
require_once __DIR__ . '/../src/VCard.php';

use JeroenDesloovere\VCard\VCard;

// define vcard
$vcard = new VCard();

// define variables
$firstname = 'Jeroen';
$lastname = 'Desloovere';
$additional = '';
$prefix = '';
$suffix = '';

// add personal data
$vcard->addName($lastname, $firstname, $additional, $prefix, $suffix);

// add work data
$vcard->addCompany('Siesqo');
$vcard->addJobtitle('Web Developer');
$vcard->addEmail('info@jeroendesloovere.be');
$vcard->addPhoneNumber(1234121212, 'PREF;WORK');
$vcard->addPhoneNumber(123456789, 'WORK');
$vcard->addAddress(null, null, 'street', 'worktown', null, 'workpostcode', 'Belgium');
$vcard->addURL('http://www.siesqo.be');

$vcard->addPhoto(__DIR__ . '/assets/landscape.jpeg');

// return vcard as a string
//return $vcard->getOutput();

// return vcard as a download
return $vcard->download();

// echo message
echo 'A personal vCard is saved in this folder: ' . __DIR__;

// or

// save the card in file in the current folder
// return $vcard->save();

// echo message
// echo 'A personal vCard is saved in this folder: ' . __DIR__;
